src/utils/in-memory-cache.ts

```diff
@@ -0,0 +1,52 @@
+import { idb } from "@/common/db";
+
+const PACMAN_WALLETS = "pacman-wallets";
+
+const inMemoryCache = () => {
+  const cache: Map<string, unknown> = new Map<string, unknown>();
+  return {
+    get: function (key: string): any {
+      return cache.get(key);
+    },
+    set: function (key: string, val: unknown) {
+      cache.set(key, val);
+    },
+  };
+};
+
+export const cache = inMemoryCache();
+
+export const updateWalletCache = async (address: string) => {
+  await saveWallet(address);
+  const walletExists: string[] = cache.get(PACMAN_WALLETS);
+  if (walletExists) {
+    walletExists.push(address);
+    cache.set(PACMAN_WALLETS, walletExists);
+  } else {
+    cache.set(PACMAN_WALLETS, [address]);
+  }
+};
+
+export const getCacheWallets = async (): Promise<string[]> => {
+  let wallets: string[] = cache.get(PACMAN_WALLETS);
```

Comment:
We should store cachedWallets as a dictionary due to O(1) lookup time

```suggestion
  let wallets: Record<string, boolean> = cache.get(PACMAN_WALLETS);
```